shared/common-adapters/avatar.native.js

```diff
@@ -1,36 +1,77 @@
 // @flow
 
 import React, {Component} from 'react'
+import {Image, TouchableWithoutFeedback} from 'react-native'
 import type {Props} from './avatar'
-import {Box, Icon} from '../common-adapters'
+import {Box} from '../common-adapters'
+import {images} from './icon.paths.native'
+import {globalStyles} from '../styles/style-guide'
+import * as shared from './avatar.shared'
 
-export default class Avatar extends Component {
-  props: Props;
+type State = {
+  avatarLoaded: boolean,
+}
 
-  state: {
-    avatarLoaded: boolean
-  };
+export default class Avatar extends Component<void, Props, State> {
+  props: Props;
+  state: State;
 
   constructor (props: Props) {
     super(props)
     this.state = {avatarLoaded: false}
   }
 
   render () {
-    return (
-      <Box style={{justifyContent: 'flex-end', ...this.props.style}} onClick={this.props.onClick}>
-        <Icon style={{...avatarStyle(this.props.size - 2)}}
-          type='placeholder-avatar' />
+    const {size} = this.props
+    const uri = {uri: shared.createAvatarUrl(this.props)}
+    console.log('uri & state & props', uri, this.state, this.props)
```

Comment:
Side note: iOS doesn't always provide a username to the tab-bar when the user is logged in. This was me trying to track the issue, but I only found it sporadically. This may warrant further investigation later down the road.